src/components/shared/RightSidebar/PostNavigation.tsx

```diff
@@ -0,0 +1,82 @@
+"use client";
+import React, { useEffect, useState } from "react";
+import RelatedPosts from "./RelatedPosts";
+import { usePathname } from "next/navigation";
+import {
+  getPost,
+  getAllOtherPosts,
+  updateRelatedPosts,
+} from "~/lib/actions/posts";
+import { INote } from "~/models/note";
+import { NoteReference } from "~/types";
+import OnThisPage from "./OnThisPage";
+
+const PostNavigation = () => {
+  const pathname = usePathname();
+  //split string to get the last part of the url
+  const splitPathname = pathname.split("/");
+  const lastPart = splitPathname[splitPathname.length - 1];
+  const [post, setPost] = useState<INote>();
+  const [relatedPosts, setRelatedPosts] = useState<NoteReference[]>();
+  const [allOtherPosts, setAllOtherPosts] = useState<NoteReference[]>();
+
+  useEffect(() => {
+    const getThePost = async () => {
+      const post = await getPost(lastPart);
```

Comment:
![image](https://github.com/dalbanhi/git-note/assets/4108484/effa3c82-ac38-4774-a1c4-8d910d83d892)

We're getting this error from passing in this complex object from the server in this server action. A quick fix would just be to parse it on the server-side first:

```js
// actions/post.ts
async function _getPost(id: string) {
  await connectToDB();

  //get the post from the database
  try {
    const post = await Note.findOne({ _id: id }).populate("relatedNotes");
    return JSON.parse(JSON.stringify(post)); // clean object before returning it
  } catch (err) {
    return null;
  }
}
```